modules/common-graphql/src/main/scala/observe/common/ObsQueriesGQL.scala

```diff
@@ -6,276 +6,720 @@ package observe.common
+                }
               }
             }
           }
-          scienceRequirements {
-            mode
-            spectroscopyRequirements {
+        }
+      }
+
+      fragment northAtomFields on GmosNorthAtom {
+        id
+        steps {
+          id
+          stepType
+          instrumentConfig {
+            exposure {
+              microseconds
+            }
+            readout {
+              ...gmosCcdMode
+            }
+            dtax
+            roi
+            grating {
+              disperser
+              order
               wavelength {
                 picometers
               }
-              resolution
-              signalToNoise
-              signalToNoiseAt {
-                picometers
-              }
-              wavelengthRange {
-                picometers
+            }
+            filter
+            fpu {
+              ... on GmosNorthBuiltinFpu {
+                builtin
               }
-              focalPlane
-              focalPlaneAngle {
-                microarcseconds
+              ... on GmosCustomMask {
+                filename
+                slitWidth
               }
-              capabilities
             }
           }
-          scienceConfiguration {
-            ... on GmosNorthLongSlit {
-              filterN:filter
-              disperserN:disperser
-              slitWidthN:slitWidth {
-                microarcseconds
+          stepConfig {
+            ...stepConfig
+          }
+        }
+      }
+
+      fragment southAtomFields on GmosSouthAtom {
+        id
+        steps {
+          id
+          stepType
+          instrumentConfig {
+            exposure {
+              microseconds
+            }
+            readout {
+              ...gmosCcdMode
+            }
+            dtax
+            roi
+            grating {
+              disperser
+              order
+              wavelength {
+                picometers
               }
             }
-            ... on GmosSouthLongSlit {
-              filterS:filter
-              disperserS:disperser
-              slitWidthS:slitWidth {
-                microarcseconds
+            filter
+            fpu {
+              ... on GmosSouthBuiltinFpu {
+                builtin
+              }
+              ... on GmosCustomMask {
+                filename
+                slitWidth
               }
             }
           }
+          stepConfig {
+            ...stepConfig
+          }
         }
       }
-    """
 
-    object Data {
-      object Observation {
-        object ScienceRequirements {
-          object SpectroscopyRequirements {
-            type Wavelength      = lucuma.core.math.Wavelength
-            type SignalToNoiseAt = lucuma.core.math.Wavelength
-            type WavelengthRange = lucuma.core.math.Wavelength
-            type FocalPlaneAngle = lucuma.core.math.Angle
-          }
+      fragment offset on Offset {
```

Comment:
Nice usage of fragments

Reply:
I just noticed the same code repeated everywhere, so I factored it out.